lib/puppet/provider/scheduled_task/win32_taskscheduler.rb

```diff
@@ -333,6 +343,17 @@ def translate_hash_to_trigger(puppet_trigger)
       self.fail "Unknown schedule type: #{puppet_trigger["schedule"].inspect}"
     end
 
+    if puppet_trigger['minutes_interval']
+      integer_interval = Integer(puppet_trigger['minutes_interval'])
+      self.fail 'minutes_interval must be an integer greater or equal to 0' unless integer_interval >= 0
+      trigger['minutes_interval'] = integer_interval
+    end
+    if puppet_trigger['minutes_duration']
+      integer_duration = Integer(puppet_trigger['minutes_duration'])
+      self.fail 'minutes_duration must be an integer greater than minutes_interval' unless integer_duration > integer_interval || (integer_duration == 0 && integer_interval == 0)
```

Comment:
similar comment here, and since there are many comparisons being made, I prefer to do:

```ruby
if integer_duration <= integer_interval && (integer_duration != 0 || integer_interval != 0)
  self.fail ...
end
```

And now that I write it that may, I'm not sure the logic checks are correct.